source/guide/index.md

```diff
@@ -6,99 +6,156 @@ order: 1.1
+
+![Native Components](./images/native-component.png)
+
+Weex implemented render engines in each end and provides a group of [built-in components](../references/components/) for basic usage. Based on these components, you can compose and wrap more custom components.
+
+Although the components in Weex look like HTML tags, you can't use all HTML tags, you can only use the built-in components and your custom components.
+
+Under the scene, Weex is using the native widgets. Although Weex emphasize the consistency of each end, we still embrace the platform's own behavior and UI differences. For example, [this `<switch>` component](http://dotwe.org/vue/d96943452b6708422197c47920903823) may look different on Android and iOS.
+
+![Different switch](./images/different-switch.png)
+
+If you want to use more native component other than the built-in components, you should have to implement it on each platform and be consistent. The most practical way is to integrate the existing native components to weex platform.
 
-You can see and edit this example at [Weex online playground](http://dotwe.org/vue/8da01827631b21150a12dd54d7114380). Use the [Weex playground app]() to scan the QR code on the right you will see the result rendered on your own phone.
+### Native Modules
 
-### Native Component
+For those features which don't rely on the UI, Weex wraps it into **modules**. You can use `weex.requireModule('xxx')` to require them. It is a way to access native capabilities in javascript, such as network, storage, clipboard, and navigator. For example, you can [use `stream` module to fetch the star count of Vue.js](http://dotwe.org/vue/2ae062b6a04124a35bbe2da3b1e5c07b).
 
-![Native Component](./images/native-component.png)
+Similarly, Weex also provides a group of [built-in modules](../references/modules/) for basic usage, and support to integrate the existing native modules to weex platform.
 
-In this example, the `<div>` and the `<text>` is rendered into corresponding native components.
+Here are some documents on how to extend native components and native modules of Weex:
+
++ [Extend Web Render](./extend-web-render.html)
++ [Extend Android](./extend-android.html)
++ [Extend iOS](./extend-ios.html)
 
 ### Write Once, Run Everywhere
 
-You can read *[How it works](../wiki/index.html)* and *[Design Principles](../wiki/design-principles.html)* to know more about the technologies and ideas behind Weex.
+Yes, Weex can build both Android, iOS and web apps with a single codebase.
 
-### More Examples
+balala.
+
+You can read *[How it works](../wiki/index.html)* and *[Design Principles](../wiki/design-principles.html)* to know more about the technologies and ideas behind Weex.
 
 ## Support Multiple Front-End Frameworks
 
-Vue.js 2.0 and Rax
+Front-end frameworks are the syntax layer of weex, they are decoupled with native render engines. Weex isn't binding with any specific front-end frameworks, instead, Weex can bring native capabilities to them.
```

Comment:
syntax layer of Weex,
they are decoupled from native render engines.
Weex does not bind with ..., instead, it brings ...